zookeeper/src/main/java/com/linecorp/armeria/server/zookeeper/ZooKeeperUpdatingListener.java

```diff
@@ -46,13 +46,31 @@ public ZooKeeperUpdatingListener(String zkConnectionStr, String zNodePath, int s
                                      Endpoint endpoint) {
         this.zkConnectionStr = requireNonNull(zkConnectionStr, "zkConnectionStr");
         this.zNodePath = requireNonNull(zNodePath, "zNodePath");
-        this.endpoint = requireNonNull(endpoint, "endpoint");
+        this.endpoint = endpoint;
```

Comment:
Could you still make this non-nullable and only allow setting `endpoint` to `null` via the constructor below?